x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_signal.ts

```diff
@@ -5,35 +5,41 @@
  */
 
 import { RulesSchema } from '../../../../common/detection_engine/schemas/response/rules_schema';
-import { INTERNAL_IDENTIFIER } from '../../../../common/constants';
 import { SignalSourceHit, Signal, Ancestor } from './types';
 
-export const buildAncestor = (doc: SignalSourceHit, rule: Partial<RulesSchema>): Ancestor => {
-  const existingSignal = doc._source.signal?.parent;
-  if (existingSignal != null) {
+/**
+ * Takes a parent signal or event document and extracts the information needed for the corresponding entry in the child
+ * signal's `signal.parents` array.
+ * @param doc The parent signal or event
+ */
+export const buildParent = (doc: SignalSourceHit): Ancestor => {
+  if (doc._source.signal != null) {
     return {
-      rule: rule.id != null ? rule.id : '',
+      rule: doc._source.signal.rule.id,
       id: doc._id,
       type: 'signal',
       index: doc._index,
-      depth: existingSignal.depth + 1,
+      // We first look for signal.depth and use that if it exists. If it doesn't exist, this should be a pre-7.10 signal
+      // and should have signal.parent.depth instead. signal.parent.depth in this case is treated as equivalent to signal.depth.
+      depth: doc._source.signal.depth ?? doc._source.signal?.parent?.depth ?? 1,
```

Comment:
Why is the `?` needed after `doc._source.signal` - seems confusing since it was checked on line 16. Should I expect that value to exist in this block of code or not?

Reply:
Good point, it's not needed - I'll remove it.